docs/spec_draft.md

```diff
@@ -1529,6 +1530,64 @@ operations correspond to [stablehlo.minimum](#stablehlominimum) and
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.collective_permute
+
+### Semantics
+
+Within each process group in the StableHLO grid, sends and receives data between
+processes in the group.
+
+The operation splits the StableHLO grid into `process_groups` as follows:
+  * `channel_id <= 0`,
+    `cross_replica(replica_groups)`.
+  * `channel_id > 0`,
+    `cross_partition(replica_groups)`.
+
+Afterwards, `result@target` is given by
+ * `operand@process_groups[i][0]`, if there exist an `i` such that  `process_groups[i][1] = target`.
+ * `result@target = T, otherwise, # where T is a zero valued tensor constant with type(T) = type(operand)`, otherwise.
```

Comment:
Proposed rewording:

```
Afterwards, `result@process` is given by:
 * `operand@process_groups[i, 0]`, if there exists an `i` such that `process_groups[i, 1] = process`.
 * `[0, ..., 0]`, otherwise.
```

Feedback:
  1. Let's use the `result@process` name, given that not every process is necessarily a target.
  2. Using `process_groups[i, 0]` notation rather than `[i][0]` because `process_groups` is a 2d tensor (so only one level of indexing is needed).
  3. Let's simplify the second bullet. If the proposed notation doesn't sound strict enough because `result` can be multi-dimensional, we can use the `broadcast_in_dim(0, [], shape(result))` notation.

Reply:
Thanks!

I would prefer `broadcast`.